java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java

```diff
@@ -114,18 +119,18 @@ public HttpClient getGridHttpClient(int connection_timeout, int socket_timeout)
         .build();
   }
 
-  private SocketConfig createSocketConfig() {
+  private SocketConfig createSocketConfig(int socketTimeout) {
     return SocketConfig.custom()
         .setSoReuseAddress(true)
-        .setSoTimeout(TIMEOUT_THREE_HOURS)
+        .setSoTimeout(socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS)
```

Comment:
My bad. I removed those checks now that they're checked in the createHttpClient() function. That looks sufficient to me.